apps/namadillo/src/slices/validators.ts

```diff
@@ -34,50 +40,116 @@ export type MyValidator = {
+  indexerVotingPower: IndexerVotingPower,
+  unbondingPeriod: bigint
+): Validator => {
+  return {
+    uuid: indexerValidator.address,
+    alias: indexerValidator.name,
+    description: indexerValidator.description,
+    address: indexerValidator.address,
+    homepageUrl: indexerValidator.website,
+    // TODO: Return this from the indexer
+    expectedApr: 0.1127,
+    unbondingPeriod: `${unbondingPeriod} days`,
+    votingPowerInNAM: BigNumber(indexerValidator.votingPower),
+    votingPowerPercentage:
+      Number(indexerValidator.votingPower) /
+      Number(indexerVotingPower.totalVotingPower),
+    commission: BigNumber(indexerValidator.commission),
+    imageUrl: indexerValidator.avatar,
+  };
+};
+
+export const allValidatorsAtom = atomWithQuery((get) => {
+  const chainParameters = get(chainParametersAtom);
+  return {
+    queryKey: ["all-validators", chainParameters.dataUpdatedAt],
+    enabled: !!chainParameters,
+    queryFn: async () => {
+      const parameters =
+        chainParameters.data?.unbondingPeriodInDays || BigInt(0);
+
+      const api = new DefaultApi();
+      const [validatorsResponse, votingPowerResponse] = await Promise.all([
+        api.apiV1PosValidatorGet(),
+        api.apiV1PosVotingPowerGet(),
+      ]);
+
+      // TODO: rename one data to items?
+      const validators = validatorsResponse.data.data;
+      const votingPower = votingPowerResponse.data;
 
-export const allValidatorsAtom = atomWithQuery((get) => ({
-  queryKey: ["all-validators"],
-  queryFn: async () => {
-    const rpcUrl = get(rpcUrlAtom);
-    const query = new Query(rpcUrl);
-    const queryResult =
-      (await query.query_all_validator_addresses()) as string[];
-    return queryResult.map(toValidator);
-  },
-}));
+      return validators.map((v) => toValidator(v, votingPower, parameters));
+    },
+  };
+});
 
 // eslint-disable-next-line
 export const myValidatorsAtom = atomWithQuery((get) => {
   const account = get(defaultAccountAtom);
+  const chainParameters = get(chainParametersAtom);
 
   // TODO: Refactor after this event subscription is enabled in the indexer
   const enablePolling = get(shouldUpdateBalanceAtom);
 
   return {
-    queryKey: ["my-validators", account.data?.address],
+    queryKey: [
+      "my-validators",
+      account.data?.address,
+      chainParameters.dataUpdatedAt,
+    ],
     enabled: account.isSuccess,
     refetchInterval: enablePolling ? 1000 : false,
     queryFn: async (): Promise<MyValidator[]> => {
-      const rpcUrl = get(rpcUrlAtom);
-      const query = new Query(rpcUrl);
-      const myValidatorsRes = await query.query_my_validators([
-        account.data?.address,
+      const unbondingPeriod =
+        chainParameters.data?.unbondingPeriodInDays || BigInt(0);
+      const api = new DefaultApi();
+      const [bondsResponse, totalVotingPowerResponse] = await Promise.all([
+        // TODO: is address always available?
+        api.apiV1PosBondAddressGet(account.data!.address!),
```

Comment:
I think we should move these two queries to their respective atoms, avoiding querying it multiple times